AirCasting/CreateSessionViews/ConnectingABView.swift

```diff
@@ -39,12 +39,21 @@ struct ConnectingABView: View {
                     isActive: $isDeviceConnected,
                     label: {
                         EmptyView()
-                    })
-            )}
+                    }
+                )
+            )
+        }
         .padding()
         .onAppear(perform: {
             bluetoothManager.centralManager.connect(selectedPeripheral,
                                                     options: nil)
+            DispatchQueue.main.asyncAfter(deadline: .now() + 10) {
+                if selectedPeripheral.state == .connecting {
+                    Log.info("Connecting to bluetooth device failed")
```

Comment:
`View` is not a great place for app logic. Please consider moving it out - be that a viewmodel or a controller-viewmodel-service combination (which would be perfect for separating concerns, code clarity and testability). Note that it will also require other logic to be moved away from here.